### PR TITLE
Rotate keys lambda logic

### PR DESCRIPTION
**Pull request by ppajersk**

### CODE REVIEW

**Comment by ppajersk** [comment]
This code will be ready for review after it's completed and rebased with pr #68

**Comment by ppajersk** [comment]
Code completed but not tested. Need to confirm a few additional cases and test all possible cases

File: awslambda/KeyManager.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+        message = (
+            f"Current access key is {self.rotate_days} days old, a new access key has been created. "
+            + f"\n - Old access key {key_id} has {self.delete_days - self.rotate_days} days left before it is expired."
+            + f"\n - New access key **Access key ID** = ({new_access_key_id}) | **Secret access key** = ({new_secret_access_key})"
+        )
+
+        # This script only runs if two key exist, which presumes 1 key is within expiring and 1 is not given the bot can do what it needs to
+        # double check that a static message like this is fine
+        self.logging.debug("Current key is 80 days old, creating new access key")
```
Comment (jlothian) [inline awslambda/KeyManager.py:173]: should the 80 here be a variable?

File: awslambda/KeyManager.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        self.send_message_to_user(user_email, message)
+
+    def delete_key(self, user_email, key_id, expired, unused, key_created_days, user):
+        access_key = user.AccessKey(key_id)
+        access_key.delete()
+
+        message = ""
+        if expired:
+            message = f"Access key ({key_id}) \n - Age: {key_created_days} days old \n - Maximum lifespan: {self.delete_days} days \n - Status: **Deleted** \n - Reason: exceeds lifespan"
+            self.logging.debug("Key is 90 days old or older, deleting key")
```
Comment (jlothian) [inline awslambda/KeyManager.py:183]: variable for days

File: awslambda/KeyManager.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        if unused:
+            message = f"Access key ({key_id}) \n - Last used: {self.warn_days} days ago \n - Maximum unused lifespan: {self.warn_days} days \n - Status: **Deleted** \n - Reason: key is not used"
+            self.logging.debug("Key has not been used in 45 days, deleting key")
```
Comment (jlothian) [inline awslambda/KeyManager.py:186]: variable for days

File: awslambda/KeyManager.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+        self.send_message_to_user(user_email, message)
+
+    def warn_user(self, user_email, expire, unused, days_to_warn, key_id):
+        message = ""
+        if expire:
+            message = f"Access key ({key_id}) \n - Age: {days_to_warn} days old \n - Expiration in: {self.delete_days - days_to_warn} days \n - Status: **Warning** \n - Reason: reaching key age limit of {self.delete_days} days \n"
+            self.logging.debug(
+                "Key age is between 80-90 days old, warning user of expiration"
```
Comment (jlothian) [inline awslambda/KeyManager.py:195]: variable for days

File: awslambda/KeyManager.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+            )
+        if unused:
+            message = f"Access key ({key_id}) \n - Last used: {days_to_warn} days ago \n - Expiration in: {self.warn_days - days_to_warn} days \n - Status: **Warning** \n - Reason: reaching key unused limit of {self.warn_days} days \n"
+            self.logging.debug(
+                "Key has not been used in 40-45 days, warning user of expiration"
```
Comment (jlothian) [inline awslambda/KeyManager.py:200]: variable for days

**Review by jlothian — CHANGES_REQUESTED** [review]

**Review by jlothian — APPROVED** [review]

**Pull request merged**